docs/microphone.rst

```diff
@@ -70,11 +119,61 @@ Functions
     * **return**: a representation of the sound pressure level in the range 0 to
       255.
 
+.. py:function:: record(duration=3000, rate=7812, wait=True)
```

Comment:
Maybe we can remove this function altogether, and just have `record_into()`.  That way memory management is explicit, it avoids situations where there's a lot of heap fragmentation, and also avoids any difficulties with this function returning an `AudioFrame` object that is growing over time.

Reply:
Sorry, I forgot the context in which AudioFrame would be growing. I take it that was the case when `wait=False`, but if we have to have a known/default value for `duration` and `rate`, wouldn't the AudioFrame be allocated at the start before the recording occurs?

I still quite like the idea of being able to do `audio.play(microphone.record(duration=3000))`.
Would it be simpler if `wait` was removed from this version and only available in `record_into()`?

Reply:
> I take it that was the case when `wait=False`, but if we have to have a known/default value for `duration` and `rate`, wouldn't the AudioFrame be allocated at the start before the recording occurs?

When `wait=False` the returned-AudioFrame will be gradually filling up.  If we go with the current implementation where the whole buffer is preallocated at the start of the recording, then I guess it's not too bad, the user just sees blank data that's gradually filling up.

So, we can keep this function.  And even support `wait=False`.  Under the hood it will simply allocate an AudioFrame of the desired size, then pass it to `record_into()`.